zerver/lib/message.py

```diff
@@ -909,9 +909,10 @@ def get_raw_unread_data(
                     "recipient__type",
                     "recipient__type_id",
                 )
+                .order_by("message_id")
             )
 
-            rows = list(reversed(user_msgs))
+            rows = list(user_msgs)
```

Comment:
But why did we have `reversed` previously? I think that might have been related to ensuring that we have the last 50k unreads, not the oldest.

Reply:
The order was:
* get user messages ordered by latest `message_id` first


```
    user_msgs = (
        UserMessage.objects.filter(
            user_profile=user_profile,
            message_id__gte=first_visible_message_id,
        )
        ...
        .order_by("-message_id")
    )
 ```
 
 * Then slice 50K of them: `cte = With(user_msgs[:MAX_UNREAD_MESSAGES])`
* Then reversed to make them in ascending order.

But reversed can make it ascending only if the `user_msgs` are still descending after the join -- which is not always the case.

Reply:
I see, got it. This makes sense.